nasty_nesting.py

```diff
@@ -0,0 +1,15 @@
+from cpmpy import *
```

Comment:
I think this is from the docs, right?

You could add it to examples/, then it will be run as a test for all solvers too.
(maybe add a comment on top saying that it is from the docs to demonstrate the free use of nested expressions?)